### DISCUSSION TITLE
커스텀 에러 코드 이 방식은 어떨까요?

### DISCUSSION THREAD

**Discussion by kmw2378** [KakaoFunding/back-end#246] [Ideas]
[Custom Error Code가 왜 필요했을까?](https://park-algorithm.tistory.com/entry/Custom-Error-Code-%EC%A0%81%EC%9A%A9%EA%B8%B0) 를 참고했습니다!

코드는 `KF` + (숫자)로 표현하면 어떨까요? 숫자는 아래와 같이 구분하면 좋을 것 같습니다!
- (도메인 식별 번호) + (에러 상황별 번호)

### 도메인별 식별 번호
|도메인|식별 번호|
|------|---|
|Auth|0|
|Product|1|
|Wish|2|

### Auth(식별번호가 0인 경우) 
|에러 상황|번호|상태 코드|
|------|---|---|
|01|엑세스 토큰이 없는 경우|404
|02|엑세스 토큰 값이 잘못된 경우|400
|02|엑세스 토큰이 만료된 경우|400
|03|엑세스 토큰 형식이 JWT가 아닌 경우|400
|04|리프레시 토큰이 없는 경우|404
|05|리프레시 토큰 값이 잘못된 경우|400
|06|리프레시 토큰이 만료된 경우|400
|07|소셜 엑세스 토큰이 없는 경우|404
|08|소셜 엑세스 토큰이 잘못된 경우|400
|09|소셜 엑세스 토큰이 만료된 경우|400
|10|소셜 리프레시 토큰이 없는 경우|404
|11|소셜 리프레시 토큰이 잘못된 경우|400
|12|소셜 리프레시 토큰이 만료된 경우|400

예를 들어 엑세스 토큰이 만료된 경우 아래와 같이 응답이 나갑니다!
```json
{
    "code" : "KF002",
    "message" : "만료된 엑세스 토큰입니다.",
    "status" : 400
}
```

지금 당장 모든 도메인 에러 상황마다 코드를 매기긴 힘드니 당장은 Auth에 대해서만 나타내면 좋을 것 같습니다!

**Comment by YeaChan05** [comment]
구체적 예외상황을 원인도 함께 명시할 수 있는 방법인거같네요
저는 좋습니다

**Reply by kmw2378** [reply]
네넵! 의견 감사합니다!
